test/micro/org/openjdk/bench/java/lang/foreign/SegmentBulkRandomFill.java

```diff
@@ -0,0 +1,183 @@
+    @Benchmark
+    public void heapSegmentFillJava() {
+        for (int i = 0; i < INSTANCES; i++) {
+            HEAP_SEGMENTS[i].fill((byte) 0);
+        }
+    }
+
+    @Fork(value = 3, jvmArgs = {"-Djava.lang.foreign.native.threshold.power.fill=0"})
+    @Benchmark
+    public void heapSegmentFillUnsafe() {
+        for (int i = 0; i < INSTANCES; i++) {
+            HEAP_SEGMENTS[i].fill((byte) 0);
+        }
+    }
+
+    @Benchmark
+    public void heapSegmentFillLoop() {
+        for (int i = 0; i < INSTANCES; i++) {
+            final long end = HEAP_SEGMENTS[i].byteSize();
+            for (long j = 0; j < end; j++) {
+                HEAP_SEGMENTS[i].set(ValueLayout.JAVA_BYTE, j, (byte) 0);
+            }
+        }
+    }
+
+    @Fork(value = 3, jvmArgs = {"-Djava.lang.foreign.native.threshold.power.fill=31"})
+    @Benchmark
+    public void nativeSegmentFillJava() {
+        for (int i = 0; i < INSTANCES; i++) {
+            NATIVE_SEGMENTS[i].fill((byte) 0);
+        }
+    }
+
+    @Fork(value = 3, jvmArgs = {"-Djava.lang.foreign.native.threshold.power.fill=0"})
+    @Benchmark
+    public void nativeSegmentFillUnsafe() {
+        for (int i = 0; i < INSTANCES; i++) {
+            NATIVE_SEGMENTS[i].fill((byte) 0);
+        }
+    }
+
+    @Benchmark
+    public void nativeSegmentFillLoop() {
+        for (int i = 0; i < INSTANCES; i++) {
+            final long end = NATIVE_SEGMENTS[i].byteSize();
+            for (long j = 0; j < end; j++) {
+                NATIVE_SEGMENTS[i].set(ValueLayout.JAVA_BYTE, j, (byte) 0);
+            }
+        }
+    }
+
+    @Fork(value = 3, jvmArgs = {"-Djava.lang.foreign.native.threshold.power.fill=31"})
+    @Benchmark
+    public void unalignedSegmentFillJava() {
+        for (int i = 0; i < INSTANCES; i++) {
+            UNALIGNED_SEGMENTS[i].fill((byte) 0);
+        }
+    }
+
+    @Fork(value = 3, jvmArgs = {"-Djava.lang.foreign.native.threshold.power.fill=0"})
+    @Benchmark
+    public void unalignedSegmentFillUnsafe() {
+        for (int i = 0; i < INSTANCES; i++) {
+            UNALIGNED_SEGMENTS[i].fill((byte) 0);
+        }
+    }
+
+    @Benchmark
+    public void unalignedSegmentFillLoop() {
+        for (int i = 0; i < INSTANCES; i++) {
+            final long end = UNALIGNED_SEGMENTS[i].byteSize();
+            for (long j = 0; j < end; j++) {
+                UNALIGNED_SEGMENTS[i].set(ValueLayout.JAVA_BYTE, j, (byte) 0);
+            }
+        }
+    }
+
+    @Fork(value = 3, jvmArgs = {"-Djava.lang.foreign.native.threshold.power.fill=31"})
+    @Benchmark
+    public void mixedSegmentFillJava() {
+        for (int i = 0; i < INSTANCES; i++) {
+            MIXED_SEGMENTS[i].fill((byte) 0);
+        }
+    }
+
+    @Fork(value = 3, jvmArgs = {"-Djava.lang.foreign.native.threshold.power.fill=0"})
+    @Benchmark
+    public void mixedSegmentFillUnsafe() {
+        for (int i = 0; i < INSTANCES; i++) {
+            MIXED_SEGMENTS[i].fill((byte) 0);
```

Comment:
loop unrolling can defeat a bit the purpose of messing up with branch misprediction here so...one solution is to avoid loop unrolling here OR

create a 
```
@CompilerControl(DONTINLINE)
static void fillSegment(MemorySegment ms, byte value) {
    memorySegment.fill(value);
}
```
which makes sure (at the cost of a very predictable call to the `fillSegment` method that:
- the inlining depth of fill is controlled and not dependent by the depth of the JMH infra caller
- the call site for fill is always the same (in term of the address in the compiled blob)